sklearn/model_selection/tests/test_validation.py

```diff
@@ -936,6 +937,138 @@ def test_cross_val_predict_with_method():
         assert_array_almost_equal(expected_predictions, predictions)
 
 
+def test_cross_val_predict_class_subset():
+    iris = load_iris()
+    X, y = iris.data, iris.target
+    X, y = shuffle(X, y, random_state=0)
+    classes = len(set(y))
+
+    # Modifies the dataset so that in a particular fold, produced by
+    # kfold.split, the training set is composed of only 2 classes instead of 3
+    y[:50] = 0
+    y[50:100] = 1
+    X[:100], y[:100] = shuffle(X[:100], y[:100], random_state=0)
```

Comment:
Could we have just done something like
```python
X = np.arange(6).reshape(3, 2)
y = [1,2,3]
cross_val_predict(LogisticRegression(), X, y, cv=3, method=method)
```
to get the same effect, i.e. broken in master and fixed here?

Even better, use
```python
X = np.arange(8).reshape(4, 2)
y = [1,1,2,3]
```
with `KFold(3)` and `KFold(4)` to create two different kinds of broken behaviour in master.